test/HeadersTest.php

```diff
@@ -885,11 +872,41 @@ public function testRequestOutWithUseProxyFalse () {
     $_SERVER['REQUEST_URI'] = $env['REQUEST_URI'];
 
     $headers = new Headers($env, $store);
-    $includePath = 'dummy';
-    $headers->requestOut($includePath);
+    $headers->requestOut();
     $this->assertEquals('en.minimaltech.co', $env['HTTP_X_FORWARDED_HOST']);
   }
 
+  public function testRequestOutUrlPatternPath () {
+    $store = $this->createStore();
+    $store->settings['url_pattern_name'] = 'path';
+    $store->settings['url_pattern_reg'] = '\/(?P<lang>[^\/.]+)(\/|\?|$)';
+
+    $env = $this->getEnv('_path');
+    $_SERVER['REQUEST_URI'] = $env['REQUEST_URI'];
+
+    $headers = new Headers($env, $store);
+    $headers->requestOut();
+    $this->assertEquals('/t.php?', $env['REQUEST_URI']);
+    $this->assertEquals('/t.php', $env['REDIRECT_URL']);
+    $this->assertEquals('/index.php', $env['HTTP_REFERER']);
+  }
+
+  public function testRequestOutUrlPatternQuery()
+  {
+    $store = $this->createStore();
+    $store->settings['url_pattern_name'] = 'query';
+    $store->settings['url_pattern_reg'] = '((\?.*&)|\?)wovn=(?P<lang>[^&]+)(&|$)';
+
+    $env = $this->getEnv('_query');
+    $_SERVER['REQUEST_URI'] = $env['REQUEST_URI'];
+
+    $headers = new Headers($env, $store);
+    $headers->requestOut();
+    $this->assertEquals('', $env['QUERY_STRING']);
+    $this->assertEquals('/t.php?wovn=ja', $env['REQUEST_URI']);
```

Comment:
why is the language still inside the request uri ?

Reply:
the code for `query` is commented out

```
            // removed this in case of redirect the lang is lost
            //$this->_env['REQUEST_URI'] = $this->removeLang($this->_env['REQUEST_URI']);
```

This is intention IMO, so I did not uncomment code

Reply:
I guess it should not affect user's site behavior so it might be fine